File: Java/[Udemy] Java POO/section13/AbstractMethods/src/entities/Circle.java
package entities;

import entities.enums.Color;

public class Circle extends Shape {

	private Double radius;

	public Circle(Color color, Double radius) {
		super(color);
		this.radius = radius;
	}

	public double getRadius() {
		return radius;
	}

	public void setRadius(double radius) {
		this.radius = radius;
	}
	
	@Override
    public double area() {
        return Math.PI * Math.pow(getRadius(), 2);
    }	
}

File: CSharp/[Udemy] CSharp POO/session8/Session100Ex4/Entities/Circle.cs
﻿using Session100Ex4.Entities.Enums;

namespace Session100Ex4.Entities
{
    internal class Circle : Shape
    {
        public double Radius { get; set; }

        public Circle(Color color, double radius) : base (color)
        {
            Radius = radius;
        }

        public override double Area()
        {
            return Math.PI * Math.Pow(Radius, 2);
        }
    }
}
